### PR TITLE
Image URL Validation

### PR DESCRIPTION
**Pull request by masalemi**
This pull request includes a commit which adds basic URL validation. This means image URLs will be checked to see if they have correct structure (outside of https:// which is assumed for recognizing image urls). In order to do this I used a urlValidator from apache commons validators and updated the pom.xml to include the needed dependencies.

### CODE REVIEW

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 1]
```diff
@@ -82,6 +83,22 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 	String replacement = "<img src=\"$1\" />";
 	String textWithImagesReplaced = userText.replaceAll(regex, replacement);
 
+	int i = 0;
+	String[] schemes = {"http","https"};
+	UrlValidator urlValidator = new UrlValidator(schemes);
+	while (true) {
```
Comment (zrolston) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: This loop should work but it's a bit hard to follow at first glance, do you think we might be able to use regex matching to extract the URLs and make it a bit more easily readable?

**Review by zrolston — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/codeu/servlets/MessageServlet.java (hunk 1 above)
Comment (masalemi) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java]: Were you thinking like creating a regex string to find URLs, adding them to a list and then iterating through the list verifying them?

Comment (zrolston) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java]: Yep that was exactly what I was thinking, using the Regex Matcher's "find" function would be a good way to structure the loop as "for each URL" which I think would be followed more easily.

Comment (masalemi) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java]: Updated to use regex! Let me know what you think.

**Review by zrolston — APPROVED** [review]
LGTM
